### PR TITLE
chore: add dev cli option to get auth token

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

DET-8850

## Test Plan

- try passing arbitrary args to curl
- try positioning the -x arg before or after the curl args
- test exit codes

Screen Shot 2023-03-03 at 11 13 00

## Commentary (optional)

Screen Shot 2023-02-21 at 15 42 26

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by hamidzr** [comment]
maybe exclude this from --help

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -9,24 +8,18 @@ repos:
         language: system
         files: '^webui/react/src/'
         types_or: [javascript, jsx, ts, tsx]
-        stages:
-          - commit
```
Comment (stoksc) [inline .pre-commit-config.yaml:13]: Why these changes?

File: harness/determined/cli/dev.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+from termcolor import colored
+
+from determined.common.api import authentication
+from determined.common.api.request import make_url
+from determined.common.declarative_argparse import Arg, Cmd
+
+
+@authentication.required
+def token(_: Namespace) -> None:
+    assert authentication.cli_auth is not None
```
Comment (stoksc) [inline harness/determined/cli/dev.py:18]: Does this give a nice message or a stack trace?

File: pre-commit/check.py [hunk 3]
```diff
@@ -54,11 +54,12 @@ def is_child(path: Path, parent: Path) -> bool:
     root
     / "proto": [
         "make fmt check build",
-        "make -C ../bindings build check",
+        "make -C ../bindings build && make -C ../webui/react check",
```
Comment (stoksc) [inline pre-commit/check.py:57]: Why these changes?

File: harness/determined/cli/dev.py (hunk 2 above)
Comment (ioga) [reply harness/determined/cli/dev.py:18]: how about we just do
`token = authentication.must_cli_auth().get_session_token()`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hamidzr) [reply .pre-commit-config.yaml:13]: hmm these are not supposed to be here :eye I'll take them out

File: pre-commit/check.py (hunk 3 above)
Comment (hamidzr) [reply pre-commit/check.py:57]: ditto

File: .pre-commit-config.yaml (hunk 1 above)
Comment (hamidzr) [reply .pre-commit-config.yaml:13]: done. I had split these up to https://github.com/determined-ai/determined/pull/6034 not sure how it ended up being reverted here.

File: harness/determined/cli/dev.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    out = output.stdout.decode("utf8")
+    try:
+        json_resp = json.loads(out)
+        if shutil.which("jq") is not None:
+            subprocess.run(["jq", "."], input=out, text=True)
+        else:
+            print(json.dumps(json_resp, indent=4))
+    except json.decoder.JSONDecodeError:
+        print(out)
+
```
Comment (hamidzr) [inline harness/determined/cli/dev.py:58]: this is basically an opportunistic prettyprinting. it's not always possible to pip the det cli output out to say jq as it might be printing other messages eg for user login or version update warnings.

**Review by loksonarius — APPROVED** [review]

**Review by stoksc — APPROVED** [review]

File: harness/determined/cli/dev.py (hunk 4 above)
Comment (hamidzr) [reply harness/determined/cli/dev.py:59]: we could probably use some shared formatter for similar purpose to print yaml, json, etc wherever we do these eg

https://github.com/determined-ai/determined/blob/9beebcc1ac86041c806c121e9bbfc33f7d14e373/harness/determined/cli/workspace.py#L117

https://github.com/determined-ai/determined/blob/9beebcc1ac86041c806c121e9bbfc33f7d14e373/harness/determined/cli/_util.py#L14

Comment (hamidzr) [reply harness/determined/cli/dev.py:59]: we have a ton of individual similar calls `ag 'dumps.*indent' harness/determined/cli/ -l`
